.github/workflows/ci.yml

```diff
@@ -0,0 +1,43 @@
+# .github/workflows/test.yml
+name: Nym CI
+
+on:
+  push:
+  pull_request:
+    types: [opened, reopened]
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        node-version: [18.14.0]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: pnpm/action-setup@v2
+        with:
+          version: 8
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'pnpm'
+      - name: Install dependencies
+        run: pnpm install
+
+      - name: Lint
+        run: pnpm lint
+
+      - name: Run tests
+        run: pnpm test
+
+      - name: Run database migrations
+        # Run migrations only on pull requests and pushes to the main branch
+        if: github.event_name == 'pull_request' || github.ref == 'refs/heads/main'
```

Comment:
Correct me if I'm wrong, but do we also want to run migrations on pushes to open pull requests?

Reply:
@cha0sg0d 
Actually yes, added the [synchronize trigger](https://github.com/orgs/community/discussions/24567) as well: d5ba46ae7292a8f5af8e0e8c01ca014296b01555

Reply:
Great, good to merge now.